src/contracts/StakingChef.sol

```diff
@@ -7,82 +7,64 @@ import "./Chef.sol";
 
 contract StakingChef is Chef {
     using SafeMath for uint256;
-    Staker[] private _stakes;
 
-    //mapping(address => mapping(address => userStakes)) public userStakes;
+    mapping(address => Staker) public staker;
 
     constructor(address metricTokenAddress) {
         setMetricToken(metricTokenAddress);
         setMetricPerBlock(4);
         toggleRewards(false);
     }
-// --------------------------------------------------------------------- staking functions
-    function stakeMetric(
-        address newAddress,
-        uint256 metricAmount,
-        uint256 newStartDate
-    ) external nonDuplicated(newAddress) {
-    if (areRewardsActive() && getTotalAllocationShares() > 0) {
+
+    // --------------------------------------------------------------------- staking functions
+    function stakeMetric(uint256 metricAmount, uint256 newStartDate) external nonDuplicated(msg.sender) {
```

Comment:
Think we should be using block.timestamp or at least do a require statement that newStartDate should be greater than block.timestamp